### PR TITLE
Feature/nested: Nested tuple support

### LINKED ISSUE #1679: Nested tuple: Support tuple as attribute type.
Support nested tuples, e.g attribute `b` in this schema.

`tuple<int32 a, tuple<int32 x, int32 y> b>`

### PR DESCRIPTION
**Pull request by markheger**
resolves #1679

### CODE REVIEW

**Comment by markheger** [comment]
`ant clean all; cd test/python/; ant test.standalone`


```
     [exec] ----------------------------------------------------------------------
     [exec] Ran 86 tests in 2930.718s
     [exec] OK (skipped=1)

BUILD SUCCESSFUL
Total time: 48 minutes 53 seconds

```

File: com.ibm.streamsx.topology/opt/python/packages/streamsx/topology/schema.py [hunk 1]
```diff
@@ -309,6 +309,15 @@ def _attribute_names(types):
 def _attribute_pytypes(types):
     pytypes = []
     for attr in types:
+        #if isinstance(attr[0], tuple):
```
Comment (ghost) [inline com.ibm.streamsx.topology/opt/python/packages/streamsx/topology/schema.py]: can be removed

File: test/python/topology/test2_namedtuple.py [hunk 2]
```diff
@@ -151,58 +436,420 @@ def test_numbers_spl_py(self):
...
+        b.slt = b.output('{[{start_time=(float64)0.1, end_time=(float64)0.2, confidence=(float64)0.5}]}')
+        s = b.stream
+        print ('-C- EXPECT ERROR CDISP9164E for unsupported type: set<list<tuple<float64 start_time,float64 end_time,float64 confidence>>>')
+        tester = Tester(topo)
+        tester.tuple_count(s, 3) # causes test to fail: type is not supported for conversion to or from Python
+        res = tester.test(self.test_ctxtype, self.test_config, assert_on_fail=False)
+        assert(False == res) # expected result: test failed
+        print ('-C- unsupported type check: PASSED')
+
+    def _test_spl_source_list_of_tuple_named_tuple_py_sink(self):
```
Comment (ghost) [inline test/python/topology/test2_namedtuple.py:493]: `_test_spl_source_list_of_tuple_named_tuple_py_sink(self):`

This test case is hidden, but should test a supported scenario.

Comment (markheger) [reply test/python/topology/test2_namedtuple.py:493]: callable returns named tuple not dict, that is the difference
restrictions to named tuple objects needs to be mentioned

**Comment by markheger** [comment]
some named tuple cases needs to reviewed in beta phase of the release

**Pull request merged**